jspwiki-main/src/main/java/org/apache/wiki/references/DefaultReferenceManager.java

```diff
@@ -301,101 +323,107 @@ private String getHashFileName( final String pageName ) {
-            if( !f.exists() ) {
-                return 0L;
-            }
+                try( final ObjectInputStream in = new ObjectInputStream( new BufferedInputStream( Files.newInputStream( f.toPath() ) ) ) ) {
+                    final StopWatch sw = new StopWatch();
+                    sw.start();
+                    LOG.debug( "Deserializing attributes for {}", p.getName() );
 
-            try( final ObjectInputStream in = new ObjectInputStream( new BufferedInputStream( Files.newInputStream( f.toPath() ) ) ) ) {
-                final StopWatch sw = new StopWatch();
-                sw.start();
-                LOG.debug( "Deserializing attributes for {}", p.getName() );
+                    final long ver = in.readLong();
+                    if( ver != serialVersionUID ) {
+                        LOG.debug( "File format has changed; cannot deserialize." );
+                        return 0L;
+                    }
 
-                final long ver = in.readLong();
-                if( ver != serialVersionUID ) {
-                    LOG.debug( "File format has changed; cannot deserialize." );
-                    return 0L;
-                }
+                    saved = in.readLong();
+                    final String name  = in.readUTF();
+                    if( !name.equals( p.getName() ) ) {
+                        LOG.debug( "File name does not match ({}), skipping...", name );
+                        return 0L; // Not here
+                    }
 
-                saved = in.readLong();
-                final String name  = in.readUTF();
-                if( !name.equals( p.getName() ) ) {
-                    LOG.debug( "File name does not match ({}), skipping...", name );
-                    return 0L; // Not here
-                }
+                    final long entries = in.readLong();
+                    for( int i = 0; i < entries; i++ ) {
+                        final String key   = in.readUTF();
+                        final Object value = in.readObject();
+                        p.setAttribute( key, value );
+                        LOG.debug( "   attr: {}={}", key, value );
+                    }
 
-                final long entries = in.readLong();
-                for( int i = 0; i < entries; i++ ) {
-                    final String key   = in.readUTF();
-                    final Object value = in.readObject();
-                    p.setAttribute( key, value );
-                    LOG.debug( "   attr: {}={}", key, value );
+                    sw.stop();
+                    LOG.debug( "Read serialized data for {} successfully in {}", name, sw );
+                    p.setHasMetadata();
                 }
-
-                sw.stop();
-                LOG.debug( "Read serialized data for {} successfully in {}", name, sw );
-                p.setHasMetadata();
             }
-        }
 
-        return saved;
+            return saved;
+        } finally {
+            lock.unlock();
```

Comment:
the diff is really clobbed up here, why `Synchronizer` can't be used here? Being a synchronized method, it shouldn't be a problem :-?